mitodl/mitxpro
# feat: associate external courseware with Django models

#### Pre-Flight checklist


**NOTE 1:** A follow-up PR will be created for this that would remove the migrated fields from the code and DB, We can't do this in this PR because the data migration would fail saying `it can't find the keys in the models because they don't exist`.

#### What are the relevant tickets?
#2581 

#### What's this PR do?
- Associates external courseware pages with Django models
- Migrates existing external courseware pages and maps their data to Django Course Models
- Adds `is_external` flag in the Django courseware pages
- Refactors the frontend code to show the appropriate external courseware pages and courseware carousels alongside

#### How should this be manually tested?
- Checkout on `master` branch
- It's fine to have seeded data from `seed_data` command
- Additionally, make sure to create a bunch of External Course and Program pages in the CMS, Ideally with different combinations of price and start dates
- Load the pages in the browser and see their behavior

**Shift to this branch**
- See the logs and notice that the newly created migration runs successfully
- Notice that new `Course, CourseRun, Program, ProgramRun, Product, ProductVersion pages get created in Django with appropriate data that you used while creating external pages in the CMS in the above section.
- Load these pages in the browser and check that everything works fine for both External and Internal courseware pages e.g. title, number of courses, enroll buttons price, dates, etc.

**NOTE 2:** This requires extensive testing to make sure that nothing breaks.

#### Where should the reviewer start?
- Setup on `master` branch and create different combinations of external courseware pages


#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
(Optional)
